content/docs/guides/proxy-exporter.md

```diff
@@ -0,0 +1,375 @@
+title: Understanding and using the proxy exporter pattern
+---
+
+# Understanding and using the proxy exporter pattern
+
+Exporters that follow the proxy exporter pattern are those who get their target from the Prometheus GET request and scrape these targets while running on a different machine via a network. In this guide, we will:
+
+* describe the proxy exporter pattern and why it is used,
+* run the [blackbox](https://github.com/prometheus/blackbox_exporter) exporter as an example of the pattern,
+* let the blackbox exporter run basic metric queries against the Prometheus [website](https://prometheus.io),
+* configure a custom query module for the blackbox exporter,
+* examine a popular pattern of configuring Prometheus to scrape exporters using relabeling.
+
+## The proxy exporter pattern?
+
+Any exporter receives requests from Prometheus and answers them with data forwarded from other systems. So technically every [exporter](/docs/instrumenting/exporters/) is a [proxy](https://en.wikipedia.org/wiki/Proxy_server). But by proxy exporter pattern we refer to a specific design, in which:
+
+* the exporter will get the target’s metrics via a network protocol.
+* the exporter does not have to run on the machine the metrics are taken from.
+* the exporter gets the targets and a query config string as parameters of Prometheus’ GET request.
+* the exporter subsequently starts the scrape after getting Prometheus’ GET requests and once it is done with scraping.
+
+This pattern is only used for certain exporters, namely the [blackbox](https://github.com/prometheus/blackbox_exporter), the [SNMP](https://github.com/prometheus/snmp_exporter) and the [IPMI exporter](https://github.com/soundcloud/ipmi_exporter).
+
+The reason is that we either can’t run the exporter on the targets, e.g. network gear speaking SNMP or servers whose OS we don’t have access too speaking IPMI, or that we are explicitly interested in the distance, e.g. latency and reachability of a website from a specific point outside of our network.
+
+## Running proxy exporters
+
+Proxy exporters are flexible regarding their environment and can be run in many ways. As regular programs, in containers, as background services, on baremetal, virtual machines. Because they are queried and do query over network they do need appropriate open ports. Otherwise they are frugal.
+
+Now let’s try it out for yourself!
+
+Use [Docker](https://www.docker.com/) to start a blackbox exporter container running by running this in a terminal:
+
+```bash
+docker run -p 9115:9115 prom/blackbox-exporter
+```
+
+You should see a few log lines and if everything went well the last one should report `msg="Listening on address"` as seen here:
+
+```
+level=info ts=2018-10-17T15:41:35.4997596Z caller=main.go:324 msg="Listening on address" address=:9115
+```
+
+## Basic querying of proxy exporters
+
+There are two ways of querying:
+
+1. Querying the exporter itself. It has its own metrics, usually available at `/metrics`.
```

Comment:
I think this is non-obvious to new users, but very important. It would be a good place to quickly explain why and link to instrumentalization.

Reply:
Good point. I added something.